app/MMB/src/scenes/tripServices/InsuranceMenuItem.js

```diff
@@ -1,20 +1,73 @@
 // @flow strict
 
 import * as React from 'react';
+import idx from 'idx';
 import { TextIcon } from '@kiwicom/mobile-shared';
-import { Translation } from '@kiwicom/mobile-localization';
+import { Translation, DateUtils, Alert } from '@kiwicom/mobile-localization';
 import { TodoMenuItem } from '@kiwicom/mobile-navigation';
+import { createFragmentContainer, graphql } from '@kiwicom/mobile-relay';
+
+import type { InsuranceMenuItem as InsuranceMenuItemType } from './__generated__/InsuranceMenuItem.graphql';
 
 type Props = {|
-  onOpenInsurance: () => void,
+  +onOpenInsurance: () => void,
+  +data: InsuranceMenuItemType,
 |};
 
-export default class InsuranceMenuItem extends React.Component<Props> {
-  render = () => {
-    const insurance = true; // TODO: fetch from the API (at least 48 hours before departure, no US nationality, CONFIRMED status only)
+export class InsuranceMenuItem extends React.Component<Props> {
+  isPastBooking = () => idx(this.props.data, _ => _.isPastBooking);
+  isStatusConfirmed = () => idx(this.props.data, _ => _.status) === 'CONFIRMED';
+
+  noUSPassengers = () => {
+    const passengers = idx(this.props.data, _ => _.passengers);
+    return (
+      passengers != null &&
+      passengers.every(passenger => {
+        const nationality = passenger && passenger.nationality;
+        return nationality != null && nationality.toLowerCase() !== 'us';
+      })
+    );
+  };
 
-    if (!insurance) {
-      // no lounges available on this trip (do not render the menu item at all)
+  getDepartureLocalTime = () => {
+    switch (this.props.data.__typename) {
+      case 'BookingOneWay':
+        return idx(this.props.data, _ => _.trip.departure.localTime);
+      case 'BookingReturn':
+        return idx(this.props.data, _ => _.outbound.departure.localTime);
+      case 'BookingMulticity':
+        return idx(this.props.data, _ => _.trips[0].departure.localTime);
+      default:
+        return null;
+    }
+  };
+
+  isMoreThan48HoursBefore = () => {
+    const departureLocalTime = this.getDepartureLocalTime();
+    return (
+      departureLocalTime &&
+      DateUtils.diffInHours(
+        new Date(departureLocalTime),
+        new Date(Date.now()),
```

Comment:
I think you should departure.utc and DateUtils.getUTCNow() instead

Reply:
`departure.utc` is not available in the GraphQL response; this is what we have in GraphQL server:
```
    time: {
      type: GraphQLDateTime,
      resolve: ({ when }: DepartureArrival): ?Date =>
        when == null ? null : when.utc, // intentional ==, can be null or undefined
    },

    localTime: {
      type: GraphQLDateTime,
      resolve: ({ when }: DepartureArrival): ?Date =>
        when == null ? null : when.local, // intentional ==, can be null or undefined
    },
```
So you'd rather I use `time` as opposed to `localTime`?

Reply:
The user could be in a different time zone 48 hours-ish before departure. That could lead to wrong comparisasion.

Reply:
Of course, I will change it!